pkg/segment/structs/segstructs.go

```diff
@@ -187,6 +187,9 @@ type OutputTransforms struct {
 	FilterRows             *BoolExpr          // discard rows failing some condition
 	MaxRows                uint64             // if 0, get all results; else, get at most this many
 	RowsAdded              uint64             // number of rows added to the result. This is used in conjunction with MaxRows.
+	Tail                   bool
+	TailRows               uint64
+	TailRequest            *TailExpr
```

Comment:
From what I can see you are only using this, `Tail bool` value just to check if the Query Aggs has a Tail Block. 
I would suggest to refactor it this way so that it will be cleaner:

remove `Tail` and move `TailRows` as part of `TailRequest` meaning it will be inside `TailExpr` struct.

So the `TailExpr` struct will look like

```
type TailExpr struct {
        TailRows            uint64
	TailRecords          map[string]map[string]interface{}
	TailPQ               *sutils.PriorityQueue
	NumProcessedSegments uint64
}
```

And the output Transforms will only have `TailRequest` and you can remove `Tail` and `TailRows`.

And when checking whether Tail Block is there in `HasTail()` you can do something like this
```
if qa != nil && qa.OutputTransforms != nil && qa.OutputTransforms.TailRequest != nil {
		return true
	}
```

Reply:
addressed in the latest commit